pkg/operator/ceph/cluster/mgr/spec.go

```diff
@@ -105,6 +105,8 @@ func (c *Cluster) makeDeployment(mgrConfig *mgrConfig) (*apps.Deployment, error)
 
 	replicas := int32(1)
 
+	k8sutil.RemoveDuplicateEnvVars(&podSpec.Spec)
```

Comment:
It seems inefficient to fix this by removing duplicates. Why not figure out why there are duplicates in the first place and fix it there?

Reply:
Good question, see a previous discussion [here](https://github.com/rook/rook/pull/6272#discussion_r489863077). Basically, it's a question of whether it's worth engineering time to track it down vs guaranteeing no duplicates with a bit extra computation.

Reply:
I'm pretty sure the var is set the second time in `cephMgrOrchestratorModuleEnvs`

Reply:
And `makeSetServerAddrInitContainer` sets it an additional time as well.

Reply:
If we have a unit test that verifies that we aren't duplicating env vars, then we don't need to use a `removeDuplicates` function on our pods. If we find duplicates, we should figure out where they are coming from and remove them. For `ROOK_POD_IP`, this is getting set to the same value so just removing one isn't an issue. But this could cover up bigger issues if an env var is set to different values multiple times. A simple remove-duplicates function can't figure out which one is actually the right one.

Reply:
Unit tests aren't guaranteed to cover all scenarios. This is especially true for OSDs there there are so many different ways to create OSDs and they all have different sets of env vars. At runtime we still need to ensure there aren't duplicates. For the mgr I would agree, let's keep the original fix that removes the duplicate. But I still vote to keep the detection to remove duplicates automatically, even with the risk of having different values. If the duplicate is removed we're more likely to find the bug than to be at the mercy of which one gets applied last at runtime.

Reply:
@BlaineEXE @travisn 

Thank you for your comments!
So, is it okey that I can keep `removeDuplicates` function as it is..? Or it is not appropriate in `mgr` case..?

Reply:
I would say remove this in the `mgr` case.

Reply:
Why not keep it? Even if there are no duplicates now, it could help prevent a future bug.